homeassistant/components/co2signal/__init__.py

```diff
@@ -13,7 +15,9 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up CO2 Signal from a config entry."""
-    coordinator = CO2SignalCoordinator(hass, entry)
+    coordinator = CO2SignalCoordinator(
+        hass, ElectricityMaps(token=entry.data[CONF_API_KEY])
```

Comment:
Can you avoid creating a new aiohttp session here and use the helper ?

https://developers.home-assistant.io/docs/integration_quality_scale_index?_highlight=aiohttp#platinum-
> Uses aiohttp or httpx and allows passing in websession (if making HTTP requests)